### ISSUE TITLE
Add support for react native new architecture 

### ISSUE THREAD

**Issue by affansk** [react-native-maps/react-native-maps#4156]
Please add support for react native new architecture

**Comment by salah-ghanim** [comment]
I'm interested in moving this forward but I think some help from the community would be awesome as well, I'm curious if anyone wants to join this effort or have experience in the new architecture already

**Comment by Simon-TechForm** [comment]
We looked into the new architecure a bit. One major blocker is [react-native docs](https://reactnative.dev/docs/new-architecture-library-intro#writing-the-javascript-spec). Using flow instead of TS isn't viable imo.

**Comment by salah-ghanim** [comment]
Good catch @Simon-TechForm I was planning on investigating it this weekend myself, but I guess we can wait for TS support, I hope it's coming soon because this lib can communicate a lot over the bridge.

**Comment by Baira** [comment]
> @Simon-TechForm it looks like TS support is available now! @salah-ghanim I can help you moving this forward. Thanks

This would be really awesome. React-native-maps is holding back our project migration to new architecture.
